src/backend/distributed/transaction/remote_transaction.c

```diff
@@ -64,6 +72,141 @@ static char *IsolationLevelName[] = {
 	"SERIALIZABLE"
 };
 
+/*
+ * These variables are necessary for running queries from a database that is not
+ * the Citus main database. Some of these queries need to be propagated to the
+ * workers and Citus main database will be used for these queries, such as
+ * CREATE ROLE. For that we create a connection to the Citus main database and
+ * run queries from there.
+ *
+ * MainDBConnection is the MultiConnection used for connecting Citus main database.
+ * IsMainDBCommand is true if this is a query in the Citus main database that is started
+ * by a query from a different database.
+ * OuterXid is the transaction id of the query from the other dtabase that started the
+ * main database query.
+ */
+MultiConnection *MainDBConnection = NULL;
+bool IsMainDBCommand = false;
+FullTransactionId OuterXid;
+bool IsMainDB = true;
```

Comment:
There should be a code comment saying why we need this boolean (i.e. it makes sense to me after reading the discussion on the PR, but we should have the conclusion as a code comment). Also @emelsimsek said it does not get set automatically for background workers. So we should mention that and probably set it ourselves for the workers that we spawn: 

>  -   CitusAuthHook is run for all client connections. But it is not run for the background workers (such as autovacuum launcher, Citus maintenance deamon). So you need to initialize isInMainDB with a default value such as isInMainDB = false;
> 
> CoordinatedTransactionCallback will be called in those background workers as well. There will be a maintenance deamon connected to MainDb. If we need to do 2PC in there, we should set isMainDB to true there.

Reply:
The default value is `true`, and when it is `true` we don't do any new logic. So, I think the default value should be fine for the background workers too. Do you think we should set it to false there?

Reply:
Good question. I think false might be better. That way if a background worker does some global DDL (very unlikely) it will always use the "connect to main database" logic. The only downside of that that I can think of would be an unnecessary connection, in the case where the background worker is actually running in the main database. 

While if the default is true, that means it will always be executed in the local database. Which would mean that if the background worker runs global DDL (again, very unlikely) while running in a non-main database then that DDL would not be propagated to other nodes.

So I think defaulting to `false` would be best. Or well, the best would be if would obviously be to automatically detect in background workers in what database we are. But I don't think that's worth the effort at the moment.

**IMPORTANT NOTE**: The reasoning above applies only when `citus.main_db` is actually set to something else than the empty string. If `citus.main_db` is not configured, I think the safest would be to keep `IsMainDB` default to `true`. The easiest way for that would be to set it to true/false from `_PG_init` based on the value of `MainDb`.

Finally, I do think that we should add a code comment that we made this decision to not worry about it for now and why:

> When citus.main_db is set IsMainDB variable is only ever set to true for connection backends, not background workers. Background worker processes will always report false here, even if they are running in a different database than the citus.main_db. This is currently fine in practice since this we only use this variable when propagating global DDL, which already seems very unlikely to happen from a background worker. And even if it did the only downside of it being true incorrectly when the background worker is running in the citus main database is that the background worker will create an unnecessary connection to the main database.